oio/rdir/client.py

```diff
@@ -106,16 +108,26 @@ def assign_all_rawx(self, max_per_rdir=None, **kwargs):
                 except KeyError:
                     self.logger.warn("rdir %s linked to rawx %s seems down",
                                      rdir_host, rawx['addr'])
-            except (NotFound, ClientException):
-                rdir = self._smart_link_rdir(rawx['addr'], all_rdir,
-                                             max_per_rdir, **kwargs)
+            except NotFound:
+                try:
+                    rdir = self._smart_link_rdir(rawx['addr'], all_rdir,
+                                                 max_per_rdir=max_per_rdir,
+                                                 **kwargs)
+                except OioException as exc:
+                    self.logger.warn("Failed to link an rdir to rawx %s: %s",
+                                     rawx['addr'], exc)
+                    continue
                 n_bases = by_id[rdir]['tags'].get("stat.opened_db_count", 0)
                 by_id[rdir]['tags']["stat.opened_db_count"] = n_bases + 1
                 rawx['rdir'] = by_id[rdir]
+            except OioException as exc:
+                self.logger.warn("Failed to check rdir linked to rawx %s "
+                                 "(thus won't try to make the link): %s",
+                                 rawx['addr'], exc)
```

Comment:
`assign_all_rawx` is used twice (https://github.com/open-io/oio-sds/blob/4.1.x/oio/cli/admin/directory.py#L127 and https://github.com/open-io/oio-sds/blob/4.1.x/oio/cli/admin/volume.py#L221). Each time, an `OioException` is catched. So, it's dead code now.